### PR TITLE
Feat: send an email about instances roles changes

### LINKED ISSUE #2037: When your role is changed by someone else you should be notified.

### PR DESCRIPTION
**Pull request by Sboonny**
- [x] I have read [Chapter's contributing guidelines](https://github.com/freeCodeCamp/chapter/blob/main/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update README.md`).
- [x] My pull request targets the `main` branch of Chapter.

Ref #2037

### CODE REVIEW

**Comment by Sboonny** [comment]
Wait, I thought both issues are the same 🤣, don't know why tho. 😅

**Review by gikf — COMMENTED** [review]
Text describing new privileges apply only to specific instance role change. For opposite change it's not right. Similarly, if more instance roles are added.

File: server/src/controllers/Users/resolver.ts [hunk 1]
```diff
@@ -72,6 +73,15 @@ export class UsersResolver {
     const oldRole = user.instance_role.name;
     if (oldRole === newRole) return user;
 
+    const emailSubject = `Your role has changed to ${newRole} in chapter's instances`;
```
Comment (Sboonny) [inline server/src/controllers/Users/resolver.ts:76]:
```suggestion
    const emailSubject = `Your role has changed to ${newRole} in the chapter's instances`;

File: cypress/e2e/dashboard/users.cy.ts [hunk 2]
```diff
@@ -43,6 +43,22 @@ describe('Users dashboard', () => {
...
+        .mhGetSubject()
+        .should(
+          'include',
+          `Your role has changed to ${memberRole} in the chapter's instances`,
+        );
+      cy.get('@email')
+        .mhGetBody()
+        .should(
+          'include',
+          `Hello, ${memberRole.name}<br />
```
Comment (Sboonny) [inline cypress/e2e/dashboard/users.cy.ts:58]:
```suggestion
          `Hello, ${memberToOwnerToMember.name}<br />

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]

File: cypress/e2e/dashboard/users.cy.ts [hunk 3]
```diff
@@ -43,6 +43,25 @@ describe('Users dashboard', () => {
...
+        .as('userName');
+      cy.waitUntilMail().mhFirst().as('email');
+
+      cy.get('@email')
+        .mhGetSubject()
+        .should('include', `Instance role changed`);
+      cy.get('@userName').then((userName) => {
+        cy.get('@email')
+          .mhGetBody()
+          .should('include', `Hello, ${userName}<br />`)
```
Comment (Sboonny) [inline cypress/e2e/dashboard/users.cy.ts:58]:
```suggestion
          .should('include', `Hello, ${userName}.<br />`)
```

**Review by ojeytonwilliams — APPROVED** [review]
LGTM 👍

**Pull request merged**